sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,647 @@
+     tC = joint_matrix_mad(sg, tA, tB, tC);
+   }
+   auto wi_data_c = matC.get_wi_data();
+   for (int i = 0; i < wi_data_c.length(); i++)
+     wi_data_c[i] *= alpha; // The indexing here "i" is in the vector owned by a WI, not in the matrix C
+   joint_matrix_store(sg, tC, memC + sg_startx * tM * N + sg_starty/SG_SIZE*tN, N, layout::row_major);
+}).wait();
+```
+
+== Query Interface
+Intel AMX, DPAS and Nvidia TPUs support different sizes and types.
+The query interface is used to validate user code and inform them about supported types, sizes, scope, and layouts by the implementation.
+This also offers development and tuning productivity by both scientists and library developers. The query interface we are proposing here is a compile-time query, 
+so there will be no runtime errors.
+The query interface proposed here consists of three functionalities:
+
+- Validation: at compile time, the validation functionality informs the user whether a specific combination is valid or not. This takes place when the user specifies all template parameters.
+
+- Default values: this provides a default shape if the user does not provide a specific combination. In this case, aliases to the `joint_matrix` type can be used, namely `joint_matrix_a/b/c` where no additional argument is needed. This form happens when the user specifies all template parameters except the sizes of the matrices (`tiles`) M, N, and K.
+
+- General query: the general query interface provides information  about sizes, types, static/dynamic, and scopes that are supported by a specific TPU implementation. This is needed to avoid padding by the user, for tuning, and efficient code generation if used by a library. The general query return an array of `combinations` of `combination` type. Each combination includes the sizes and the types for the matrices A, B, and C. Note that for each TPU, the query returns `max_msize, max_nsize, max_ksize` or `msize, nsize, ksize` exclusively depending whether the implementation supports a continuous or discrete number of sizes. For example, Intel AMX implementation supports a continuous number of sizes so the `max_*` variant is applied and only the maximum number is returned. DPAS implementation, on the other hand, supports a discrete list of numbers so the  `msize, nsize, ksize` variant is applied.  This form takes place when users only specify the TPU they are interested in using.
+
+The table below provides a description for each of the member variables and type aliases in `tpu_params` class and the forms in which  they are defined.
+
+[frame="none",options="header"]
+|======================
+| Member/type alias in `tpu_params` | Forms they are defined in |Description
+|`type_a`| validation, default values|type alias for the type of matrix A
+|`type_b`|  validation, default values|type alias for the type of matrix B
+|`type_c`|  validation, default values|type alias for the type of matrix C
```

Comment:
Should this be `type_accumulator` for consistency with `matrix::use`?

Reply:
This is a good question. @gmlueck, what do you think?